### PR TITLE
feat: error handling for function provider

### PR DESCRIPTION
**Pull request by bharatnpti**
Added error handling to the provide method in the class, throwing a NoSuchElementException when no matching function is found for the provided function name.

### CODE REVIEW

File: arc-agents/src/main/kotlin/functions/LLMFunctionProvider.kt [hunk 1]
```diff
@@ -31,8 +31,20 @@ class CompositeLLMFunctionProvider(
...
 
+    /**
+     * Retrieves a list of LLMFunctions matching the given function name.
+     *
+     * @param functionName The name of the function to search for.
+     * @return List of LLMFunctions matching the function name.
+     * @throws NoSuchElementException if no matching LLMFunction is found.
+     */
+
     override fun provide(functionName: String): List<LLMFunction> {
```
Comment (patwlan) [inline arc-agents/src/main/kotlin/functions/LLMFunctionProvider.kt:42]: We can change this to 
`override fun provide(functionName: String): LLMFunction?`
`override fun provide(functionName: String): Result<LLMFunction, FunctionNotFoundException>`
The list only made sense when we were providing functions by their group name.

File: arc-agents/src/main/kotlin/functions/LLMFunctionProvider.kt [hunk 2]
```diff
@@ -31,8 +31,20 @@ class CompositeLLMFunctionProvider(
...
-        return functions().filter { it.name == functionName }
+        val matchingFunctions = functions().filter { it.name == functionName }
+        if (matchingFunctions.isEmpty()) {
+            throw NoSuchElementException("No matching LLMFunction found for name: $functionName")
```
Comment (patwlan) [inline arc-agents/src/main/kotlin/functions/LLMFunctionProvider.kt:45]: Can we make this an explicit type, for example, FunctionNotFoundException?

File: build.gradle.kts [hunk 3]
```diff
@@ -17,7 +17,7 @@ plugins {
 
 subprojects {
     group = "io.github.lmos-ai.arc"
-    version = "0.24.0"
+    version = "0.25.0"
```
Comment (patwlan) [inline build.gradle.kts:20]: We can use the version 0.24.0. It has not been released yet.

File: arc-agents/src/main/kotlin/functions/LLMFunctionProvider.kt (hunk 2 above)
Comment (bharatnpti) [reply arc-agents/src/main/kotlin/functions/LLMFunctionProvider.kt]: done

File: arc-agents/src/main/kotlin/functions/LLMFunctionProvider.kt (hunk 1 above)
Comment (bharatnpti) [reply arc-agents/src/main/kotlin/functions/LLMFunctionProvider.kt]: done

File: build.gradle.kts (hunk 3 above)
Comment (bharatnpti) [reply build.gradle.kts:20]: done

**Pull request merged**